pkg/configdb/configdb.go

```diff
@@ -0,0 +1,77 @@
+package configdb
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path"
+	"sync"
+)
+
+var mu sync.Mutex
+
+type GlobalOptions struct {
+	IP                 string
+	Gateway            string
+	LeaseDuration      int
+	AuthToken          string
+	ListenInterface    string
+	HTTPListenAddress  string
+	TLSEnabled         bool
+	TLSPrivateKeyFile  string
+	TLSCertificateFile string
+	Options            interface{}
+}
+
+type Clients struct {
+	Clients []Client
+}
+
+type Client struct {
+	Hostname string
+	MAC      string
+	IP       string
+	Options  interface{}
+}
+
+var DB *Clients
+var Config *GlobalOptions
+
+func Init(config_path string) error {
+	content, err := os.ReadFile(path.Join(config_path, "rest-dhcpd-clients.json"))
+	if err != nil {
+		log.Printf("%s", err)
+		return err
```

Comment:
Please don't `log` all over the place but prefer to return wrapped errors with context. In this case, I suggest doing: `return fmt.Errorf("reading %s: %w", path.Join(config_path, "rest-dhcpd-clients.json"), err)`.You can put the full path into a separate variable to facilitate re-using the same formatting logic.

Reply:
Fixed